### PR TITLE
Add Facebook authorization

### PR DESCRIPTION
**Pull request by LanchevychMaxym**

### CODE REVIEW

File: EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs [hunk 1]
```diff
@@ -211,6 +212,13 @@ public interface IAuthService
         /// <returns>Returns Google user information</returns>
         /// 
         Task<UserDTO> GetGoogleUserAsync(string providerToken);
-       
+
+        /// <summary>
+        /// Sign in using Facebook
+        /// </summary>
+        /// <param name="facebookUser"></param>
+        Task<UserDTO> FacebookLogin(FacebookUserInfo facebookUser);
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs:220]:
```suggestion
        Task<UserDTO> FacebookLoginAsync(FacebookUserInfo facebookUser);
```

**Review by tetyanakutsko — APPROVED** [review]

**Review by voroniak — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs (hunk 1 above)
Comment (LanchevychMaxym) [reply EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs]: Thanks, I've fix it

**Review by Yuriyhp — APPROVED** [review]

**Review by MykhailoKlapchuk — APPROVED** [review]

File: EPlast/EPlast.WebApi/Controllers/AuthController.cs [hunk 2]
```diff
@@ -96,7 +97,14 @@ public async Task<IActionResult> Login([FromBody] LoginDto loginDto)
         [AllowAnonymous]
         public IActionResult GoogleClientId()
         {
-            return Ok(new { id = ConfigSettingLayoutRenderer.DefaultConfiguration.GetSection("GoogleAuthentication")["GoogleClientId"] });
+            return Ok(new { id = ConfigSettingLayoutRenderer.DefaultConfiguration.GetSection("GoogleAuthentication")["GoogleClientId"]});
+        }
+
+        [HttpGet("FacebookAppId")]
+        [AllowAnonymous]
+        public IActionResult FacebookAppId()
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/AuthController.cs:105]:
```suggestion
        public IActionResult GetFacebookAppId()
```

File: EPlast/EPlast.BLL/Models/FacebookUserInfo.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        public string DataAccessExpirationTime { get; set; }
+        public string Email { get; set; }
+        public string ExpiresIn { get; set; }
+        public string GraphDomain { get; set; }
+        public string Id { get; set; }
+        public string Name { get; set; }
+        public object Picture { get; set; }
+        public string UserId { get; set; }
+        public string SignedRequest { get; set; }
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Models/FacebookUserInfo.cs:19]:
```suggestion
```

File: EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs [hunk 4]
```diff
@@ -211,6 +212,13 @@ public interface IAuthService
...
         /// 
         Task<UserDTO> GetGoogleUserAsync(string providerToken);
-       
+
+        /// <summary>
+        /// Sign in using Facebook
+        /// </summary>
+        /// <param name="facebookUser"></param>
+        Task<UserDTO> FacebookLoginAsync(FacebookUserInfo facebookUser);
+
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Interfaces/Auth/IAuthService.cs:221]:
```suggestion
```

File: EPlast/EPlast.WebApi/Controllers/AuthController.cs [hunk 5]
```diff
@@ -123,6 +131,33 @@ public async Task<IActionResult> GoogleLogin(string googleToken)
...
+
+                return Ok(new { token = generatedToken });
+            }
+            catch (Exception exc)
+            {
+                _loggerService.LogError(exc.Message);
+            }
+
+            return BadRequest();
+
```
Comment (voroniak) [inline EPlast/EPlast.WebApi/Controllers/AuthController.cs:159]:
```suggestion
                var user = await _authService.FacebookLoginAsync(userInfo);
                if (user == null)
                {
                    return BadRequest();
                }
                var generatedToken = await _jwtService.GenerateJWTTokenAsync(user);

                return Ok(new { token = generatedToken });

```
Don't we have default error handling implemented?

**Review by voroniak — CHANGES_REQUESTED** [review]

**Review by voroniak — APPROVED** [review]
LGTM!

**Pull request merged**